integrations/oci/sdk/runtime/src/main/java/io/helidon/integrations/oci/sdk/runtime/OciConfigBean.java

```diff
@@ -0,0 +1,347 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.oci.sdk.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import io.helidon.builder.config.ConfigBean;
+import io.helidon.config.metadata.ConfiguredOption;
+import io.helidon.config.metadata.ConfiguredValue;
+
+import com.oracle.bmc.auth.ConfigFileAuthenticationDetailsProvider;
+
+import static io.helidon.integrations.oci.sdk.runtime.OciAuthenticationDetailsProvider.ALL_STRATEGIES;
+import static io.helidon.integrations.oci.sdk.runtime.OciAuthenticationDetailsProvider.AuthStrategy;
+import static io.helidon.integrations.oci.sdk.runtime.OciAuthenticationDetailsProvider.TAG_AUTO;
+import static io.helidon.integrations.oci.sdk.runtime.OciAuthenticationDetailsProvider.TAG_CONFIG;
+import static io.helidon.integrations.oci.sdk.runtime.OciAuthenticationDetailsProvider.TAG_CONFIG_FILE;
+import static io.helidon.integrations.oci.sdk.runtime.OciAuthenticationDetailsProvider.TAG_INSTANCE_PRINCIPALS;
+import static io.helidon.integrations.oci.sdk.runtime.OciAuthenticationDetailsProvider.TAG_RESOURCE_PRINCIPALS;
+
+/**
+ * Configuration used by {@link OciAuthenticationDetailsProvider}.
+ *
+ * @see OciExtension
+ */
+// note: this is intended to be a replica to the properties carried from the cdi integrations previously done for MP
+@ConfigBean(OciConfigBean.NAME)
+public interface OciConfigBean {
+
+    /**
+     * The config is expected to be under this key.
+     */
+    String NAME = "oci";
+
+    /** primary hostname of metadata service. */
+    String IMDS_HOSTNAME = "169.254.169.254";
+    /** primary base url of metadata service. */
+    String PRIMARY_IMDS_URL = "http://" + IMDS_HOSTNAME + "/opc/v2/";
+
+    /** fallback base url of metadata service. */
+    String FALLBACK_IMDS_URL = "http://" + IMDS_HOSTNAME + "/opc/v1/";
+
+    /**
+     * The singular authentication strategy to apply. This will be preferred over {@link #authStrategies()} if both are
+     * present.
+     *
+     * @return the singular authentication strategy to be applied
+     */
+    @ConfiguredOption(allowedValues = {
+            @ConfiguredValue(value = TAG_AUTO, description = "auto select first applicable"),
+            @ConfiguredValue(value = TAG_CONFIG, description = "simple authentication provider"),
+            @ConfiguredValue(value = TAG_CONFIG_FILE, description = "config file authentication provider"),
+            @ConfiguredValue(value = TAG_INSTANCE_PRINCIPALS, description = "instance principals authentication provider"),
+            @ConfiguredValue(value = TAG_RESOURCE_PRINCIPALS, description = "resource principals authentication provider"),
+    })
+    Optional<String> authStrategy();
+
+    /**
+     * The list of authentication strategies that will be attempted by
```

Comment:
"strategies" -> "strategy descriptors" if you're staying with the terminology you direct the reader to